Server/src/controllers/DataSetController.ts

```diff
@@ -0,0 +1,245 @@
+     */
+
+    /**
+     * This controller will take a request, send it to the getDataService to acquire an array 
+     * containing the unapproved data sets in the database
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForUnapprovedDatsets(request: Request, response: Response) {
+        try {
+            this.dataSetService = new DataSetService();
+            let requestResponse = await this.dataSetService.getUnapprovedAllDatasets()
+            return response.status(requestResponse.statusCode).json(requestResponse.message);
+        } catch (error) {
+            response.status(error.status).json(error.message);
+        }
+    }
+
+    /**
+     * This method verifies that a request has at least one valid search parameter and returns true if so
+     * and false if not.
+     * 
+     * @param request 
+     * An object containing the request information and parameters: Request
+     */
+    private validateInputData(request: Request) {
+        if (request.query.hasOwnProperty('datasetId') || request.query.hasOwnProperty('material')
+            || request.query.hasOwnProperty('year') || request.query.hasOwnProperty('categoryId')
+            || request.query.hasOwnProperty('lastName')) { return true }
+        else {
+            return false
+        }
+    }
+
+    async createRequestToFlagDataset(request: Request, response: Response) {
+        if (!request.query && !request.query.datasetId) {
+            response.status(400).json("No datasetID provided to flag dataset");
+        }
+        else {
+            let datasetIdToFlag = Number(request.query.datasetId)
+            let flaggedComments = String(request.query.flaggedComments)
+            let additionalComments = String(request.query.additionalComments)
+            try {
+                let requestResponse = await this.dataSetService.flagNewDataset(datasetIdToFlag, flaggedComments, additionalComments)
+                return response.status(requestResponse.statusCode).json(requestResponse.message);
+            } catch (error) {
+                response.status(error.status).json(error.message);
+            }
+        }
+    }
+
+    async createRequestToRejectDataset(request: Request, response: Response) {
+        let requestParam = request.params.datasetId;
+        let datasetId = Number(requestParam);
+        if (isNaN(datasetId)) {
+            response.status(400).json("Invalid data set ID entered");
+        }
+        else {
+            try {
+                this.dataSetService = new DataSetService();
+                let requestResponse = await this.dataSetService.rejectDataSet(datasetId)
+                return response.status(requestResponse.statusCode).json(requestResponse.message);
+            } catch (error) {
+                response.status(error.status).json(error.message);

```

Comment:
so here we are replying with errors.status and json(error.message) this looks good, then the above try catch should be the same, might as well just put the try catch in a separate funciton and wrap the core code in it like so:
withTryCatch(() => yourcode())
const withTryCatch(delegate) => {
  try {
    delegate()
  }
  catch(error)  {
    your typical handler
  }
}

Reply:
Good idea